File: editor/en/[260]Single Number III.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function (nums) {
    let map = new Map();
    let arr = [];

    for (const num of nums) {
        map.set(num, (map.get(num) || 0) + 1);
    }

    for (const key of map.keys()) {
        if (map.get(key) === 1) {
            arr.push(key);
        }
    }

    return arr;
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[260]Single Number III.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int[] singleNumber(int[] nums) {
        int xor = 0;

        for (int num : nums) {
            xor ^= num;
        }

        int bit =xor &-xor;
        int[] result = new int[2];

        for(int num : nums) {
            if((num & bit) == 0) {
                result[0] ^= num;
            }else{
                result[1] ^= num;
            }
        }

        return result;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
/*
class Solution {
    public int[] singleNumber(int[] nums) {
        int bitmask = 0;
        for (int num : nums) {
            bitmask ^= num;
        }
        int diff = bitmask & (-bitmask);
        int x = 0;
        for (int num : nums) {
            if ((num & diff) != 0) {
                x ^= num;
            }
        }
        return new int[]{x, bitmask ^ x};
    }
}
 */